app/src/main/java/solendsdkfacade/mapper/SolendMethodResultMapper.kt

```diff
@@ -0,0 +1,31 @@
+package solendsdkfacade.mapper
+
+import com.google.gson.Gson
+import org.p2p.wallet.utils.fromJsonReified
+import solendsdkfacade.model.SolendMethodResultError
+import solendsdkfacade.model.SolendMethodResultSuccess
+import timber.log.Timber
+
+class SolendMethodResultMapper(
+    // public due to inline
+    val gson: Gson
+) {
+
+    inline fun <reified SuccessType : SolendMethodResultSuccess> fromSdk(sdkResult: String): SolendResult<SuccessType> {
+        return SolendResult(
+            success = gson.fromJsonOrNull(sdkResult),
```

Comment:
works like`kotlin.Result` class
try parsing the string as success class and error class